docs/development/maintainers.md

```diff
@@ -108,3 +108,16 @@ final number. Follow the relevant steps from {ref}`release-instructions`.
 
 Cherry pick the corresponding documentation commits to the `stable` branch. Use
 `[skip ci]` in the commit message.
+
+### Upgrading pyodide to a new version of CPython
+
+For example: `v3.11.1` -> `v3.11.2`
+
+1. Download the **Gzipped source tarball** at https://www.python.org/downloads/release/python-3112 into `downloads/`
+2. `shasum -a 256 downloads/Python-3.11.2.tgz > cpython/checksums`
+3. `git grep --name-only "3.11.1" ` # All these files will need to be updated.
+4. After updating the Python version in `Dockerfile`, create a new Docker image.
+5. Upload the new Docker image to https://hub.docker.com/r/pyodide/pyodide-env
```

Comment:
@hoodmane @rth
Could one of you please do steps 4 and 5 using the `Dockerfile` above so that I can complete the process?

Reply:
I triggered the GHA that builds and published the image in https://github.com/pyodide/pyodide/actions/runs/4306835957 So you would need to update the URL once that completes, assuming it passes.

BTW, do we currently need to push the branch to upstream to run that action from a PR? That's what I had to do at least.

Reply:
> do we currently need to push the branch to upstream to run that action from a PR?

If you (someone who has write access to the repo) goes to https://github.com/pyodide/pyodide/actions/workflows/docker_image.yml you should see `Run workflow` because of this line https://github.com/pyodide/pyodide/blob/main/.github/workflows/docker_image.yml#L3

I (someone who does not have write access to the repo) will not see the `Run workflow` button.
* https://docs.github.com/en/actions/using-workflows/events-that-trigger-workflows#workflow_dispatch

Reply:
I updated the task list and will now push with [`image: 20230301-chrome109-firefox109-py311`](https://hub.docker.com/layers/pyodide/pyodide-env/20230301-chrome109-firefox109-py311/images/sha256-af1602aea1534fe244d649ab222f4b7b8fe30208e6a63813b502259b55f48d8e?context=explore)

Reply:
yes, took me a few trials to get a reasonably generated tag. Thanks!

Reply:
Arrgghh...  Tests are still saying the CircleCI image is Py3.11.1, not 3.11.2.

Reply:
Sorry for the confusion, I should have just built it locally. Will try again and force-push the image so you don't have to change anything.

Reply:
Yeah, we need to merge this first and then run the workflow to use the updated Dockerfile, which is a bit annoying.

Reply:
Yes.  There is a step in the Docker build process that I am not yet seeing.

[`.github/workflows/docker_image.yml`](https://github.com/pyodide/pyodide/blob/main/.github/workflows/docker_image.yml) creates the Docker image [`20230301-chrome97-firefox96`](https://hub.docker.com/layers/pyodide/pyodide-env/20230301-chrome97-firefox96/images/sha256-d8ae72579718b2cdc505207f2fb18682f91073ca323e91f71aad5486190858ed?context=explore) and pushes it up to Docker Hub but the image that this process finally uses is [`20230301-chrome109-firefox109-py311`](https://hub.docker.com/layers/pyodide/pyodide-env/20230301-chrome109-firefox109-py311/images/sha256-8a9857ee88bd97910b8b7e539ed401d8b40b2deeff6796b8db7583d92ec25243?context=explore).  How does the latter get built?

Should `.github/workflows/docker_image.yml` take a `python_version` as input in addition to the `chrome_version` and `firefox_version` that it already takes and then build the final Docker image?  Alternatively, it could read the Python version from `Makefile.envs` like `.github/workflows/main.yml` does.  Or, perhaps there are reasons for not building the final image in a GitHub Action.